### PR TITLE
c-deps: fix RocksDB on Windows without assertions

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

**Comment by benesch** [comment]
Queued another run. Failed test seemed flaky.

**Review by benesch — APPROVED** [review]
LGTM. We'll see if they have complaints about the duplication upstream anyway.

File: c-deps/rocksdb-0001-Fix-NDEBUG-windows.patch [hunk 1]
```diff
@@ -0,0 +1,90 @@
+diff --git a/port/win/env_win.cc b/port/win/env_win.cc
+index 8099f84..1e4a917 100644
+--- a/port/win/env_win.cc
++++ b/port/win/env_win.cc
+@@ -1077,13 +1077,19 @@ std::string Env::GenerateUniqueId() {
+   UuidCreateSequential(&uuid);
+ 
+   RPC_CSTR rpc_str;
+-  auto status = UuidToStringA(&uuid, &rpc_str);
+-  assert(status == RPC_S_OK);
```
Comment (petermattis) [inline c-deps/rocksdb-0001-Fix-NDEBUG-windows.patch]: Was there a problem with a warning about unused variables? I think you can do:

```
  (void)status;
```

To silence unused warnings.

Comment (benesch) [inline c-deps/rocksdb-0001-Fix-NDEBUG-windows.patch]: @tamird liked this way better

Comment (petermattis) [inline c-deps/rocksdb-0001-Fix-NDEBUG-windows.patch]: The duplicate calls are a mild maintainability concern. Easy to change one without changing the other. And sprinkling `#ifdef NDEBUG` is a bit unsightly.

**Comment by tamird** [comment]
We don't care here, do we? We can bikeshed it upstream.

**Comment by petermattis** [comment]
> We don't care here, do we? We can bikeshed it upstream.

No, I don't care a wit. Just anticipating for you.

**Pull request merged**